framework/integration/basic_test.go

```diff
@@ -0,0 +1,85 @@
+// +build k8srequired
+
+package integration
+
+import (
+	"reflect"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Test_Finalizer_Integration_Basic is a integration test for basic finalizer
+// operations. The test verifies that finalizers are added and removed as
+// expected. It does not cover correct behavior with reconciliation.
```

Comment:
IMO there integration should be an integration test to ensure reconciliation for delete events is replayed. This is the whole point why we are doing all this and we have to be very certain this is guaranteed. Now, and in the future. I would say we get this PR through and have another PR directly after to add the other test.

Reply:
💯

Reply:
I agree on adding another testcase, I could do it in this PR or another one, no preference from my side.